packages/http-server/src/types.ts

```diff
@@ -0,0 +1,102 @@
+import {AddressInfo} from 'net';
+import * as http from 'http';
+import * as https from 'https';
+import {IncomingMessage, ServerResponse} from 'http';
+
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/http-server
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+export type RequestListener = (
+  req: IncomingMessage,
+  res: ServerResponse,
+) => void;
+
+/**
+ * Basic HTTP server listener options
+ *
+ * @export
+ * @interface ListenerOptions
+ */
+export interface ListenerOptions {
+  host?: string;
+  port?: number;
+}
+
+/**
+ * HTTP server options
+ *
+ * @export
+ * @interface HttpOptions
+ */
+export interface HttpOptions extends ListenerOptions {
+  protocol?: 'http';
+}
+
+/**
+ * HTTPS server options
+ *
+ * @export
+ * @interface HttpsOptions
+ */
+export interface HttpsOptions extends ListenerOptions, https.ServerOptions {
+  protocol: 'https';
+}
+
+/**
+ * HTTP/2 server options
+ *
+ * @export
+ * @interface Http2Options
+ */
+export interface Http2Options extends ListenerOptions {
+  protocol: 'http2';
```

Comment:
The built-in `http2` module support HTTP/2 protocol over plain unencrypted socket (similar to HTTP) or over TLS socket (similar to HTTPS). How do you envision encoding this configuration flag when there is a single protocol value `http2`?